packages/foam-vscode/src/core/utils/path.ts

```diff
@@ -0,0 +1,102 @@
+import { CharCode } from '../common/charCode';
+import { posix } from 'path';
+import { promises, constants } from 'fs';
+
+export function asPath(value: string): [string, string] {
```

Comment:
I am confused by this fn, is this `fromFsPath`? I can't seem to find that

Reply:
let's add documentation for these functions, let's help future developers working on this codebase :)

Reply:
I've wrote about `fromFsPath` indeed but later delete the comment as part of some clean up before requesting revision, so as to not overwhelm you yet more. `fromFsPath` is `asPath`, the entry point to the module (originally `toUriPath`). The problem with `fromFsPath` is that the argument is not actually a filesystem path since the same function is used as an entry point both from paths encoded in URLs (for example, in order to normalize the drive letter) and from filesystem paths. This was also the case for the original `toUriPath`. So there is an asymmetry that works against having the pair `fromFsPath / toFsPath`.

Reply:
I've introduced the `fromFsPath` name after all.